server/controllers/finance/reports/cashReport/index.js

```diff
@@ -0,0 +1,294 @@
+/**
+ * cashReport Controller
+ *
+ *
+ * This controller is responsible for processing cash report.
+ *
+ * @module finance/cashReport
+ *
+ * @requires lodash
+ * @requires lib/db
+ * @requires lib/ReportManager
+ * @requires lib/errors/BadRequest
+ */
+
+const _ = require('lodash');
+const db = require('../../../../lib/db');
+const ReportManager = require('../../../../lib/ReportManager');
+const BadRequest = require('../../../../lib/errors/BadRequest');
+
+const TEMPLATE_1 = './server/controllers/finance/reports/cashReport/report_combined.handlebars';
+const TEMPLATE_2 = './server/controllers/finance/reports/cashReport/report_separated.handlebars';
+
+// expose to the API
+exports.document = document;
+
+function getOpeningBalance(accountId, maxDate) {
+  const sql = `
+  SELECT
+    IFNULL(SUM(g.debit_equiv - g.credit_equiv), 0) AS openingBalance
+  FROM 
+ 	  general_ledger AS g
+  WHERE
+	  g.account_id = ? AND
+	  g.trans_date < DATE(?);`;
+
+  return db.one(sql, [accountId, maxDate]);
+}
+
+
+function getRecordQuery(token, format, openingBalance) {
+  let query;
+
+  if (format === 1) {
+    query =
+      `
+    SET @cbal := ${openingBalance};
```

Comment:
MySQL user-defined variables like `@cbal` are shared across connections.  This was the reason for bug https://github.com/IMA-WorldHealth/bhima-2.X/issues/1273.  If two reports are created simultaneously, the `@cbal` of one will destroy the `@cbal` of the other and a user will get an unpredictable result.  I worry this will have the same results.

The way to guarantee this will not occur is to initialize the logic defining `@cbal` _inside_ the SQL statement.  That way, `@cbal` is defined at the same time as it is used.  See https://github.com/IMA-WorldHealth/bhima-2.X/issues/1273 that fixed the above issue.